db/model/generator.js

```diff
@@ -167,45 +181,21 @@ module.exports = function generator(seq, dataTypes) {
        * @returns {Promise} created generator with collectors (if any)
        */
       createWithCollectors(requestBody, whereClauseForNameInArr) {
-        // reject the request if requestBody.collectors contain duplicate names
-        if (common.checkDuplicatesInStringArray(requestBody.collectors)) {
-          const err = new dbErrors.DuplicateCollectorError();
-          err.resourceType = 'Collector';
-          err.resourceKey = requestBody.collectors;
-          return Promise.reject(err);
-        }
-
-        const options = {};
-        let generatorId;
+        let createdGenerator;
         let collectors; // will be populated with actual collectors
-        options.where = whereClauseForNameInArr(requestBody.collectors || []);
         return new seq.Promise((resolve, reject) =>
-          seq.models.Collector.findAll(options)
+          utils.validateCollectors(seq, requestBody.collectors,
+            whereClauseForNameInArr)
           .then((_collectors) => {
-
-            /*
-             * If requestBody does not have a collectors field, OR
-             * if the number of collectors in requestBody MATCH the
-             * GET result, order the collectors AND create the generator.
-             * Else throw error since there are collectors that don't exist.
-             */
-            if (!requestBody.collectors ||
-              (_collectors.length === requestBody.collectors.length)) {
-              collectors = _collectors;
-              return Generator.create(requestBody);
-            }
-
-            const err = new dbErrors.ResourceNotFoundError();
-            err.resourceType = 'Collector';
-            err.resourceKey = requestBody.collectors;
-            throw err;
-          }) // if successful create, add collectors
-          .then((createdGenerator) => {
-            generatorId = createdGenerator.id;
-            return createdGenerator.addCollectors(collectors);
+            collectors = _collectors;
+            return Generator.create(requestBody);
+          })
+          .then((_createdGenerator) => {
+            createdGenerator = _createdGenerator;
+            generatorId = _createdGenerator.id;
```

Comment:
generatorId is not defined. Probably should turn on strict mode in this file, so that the tests can catch this type of mistakes.